api/src/main/java/org/openmrs/module/fhir2/api/util/FhirUtils.java

```diff
@@ -140,6 +146,33 @@ public static Optional<OpenmrsEncounterType> getOpenmrsEncounterType(Encounter e
 		}
 	}
 	
+	/*
+	 * Determines the condition type based on the given category:
+	 * 1. "encounter-diagnosis" → Diagnosis
+	 * 2. "problem-list-item" → Condition
+	 * 3. Any other category → returns empty
+	 *
+	 * Special case:
+	 * - If category is empty, returns Condition.
+	 *
+	 * @param condition The condition object to evaluate.
+	 * @return The condition type: "Diagnosis", "Condition", or empty.
+	 */
+	public static Optional<OpenmrsConditionType> getOpenmrsConditionType(Condition condition) {
+		if (condition.hasCategory() && condition.getCategoryFirstRep().hasCoding()) {
+			String code = condition.getCategoryFirstRep().getCodingFirstRep().getCode();
```

Comment:
Rather than relying on the first rep, we should iterate through the possible options. The terminology binding here is "extensible" and it's a `0..*` binding, so valid FHIR resources can communicate other values here and we should process those and only error out if there is no code from the system http://terminology.hl7.org/CodeSystem/condition-category. We do something similar with `OpenmrsEncounterType` above.